airbyte-integrations/connectors/destination-iceberg-v2/src/main/kotlin/io/airbyte/integrations/destination/iceberg/v2/io/IcebergUtil.kt

```diff
@@ -176,44 +182,83 @@ class IcebergUtil {
+            }
+
+        // Common S3/Iceberg properties shared across all catalog types.
+        // The S3 endpoint is optional; if provided, it will be included.
+        val s3CommonProperties =
+            mutableMapOf<String, String>(
+                    CatalogProperties.FILE_IO_IMPL to S3FileIO::class.java.name,
+                    S3FileIOProperties.ACCESS_KEY_ID to awsAccessKeyId,
+                    S3FileIOProperties.SECRET_ACCESS_KEY to awsSecretAccessKey,
+                    // Required for MinIO or other S3-compatible stores using path-style access.
+                    S3FileIOProperties.PATH_STYLE_ACCESS to "true"
+                )
+                .apply {
+                    config.s3BucketConfiguration.s3Endpoint?.let { endpoint ->
+                        this[S3FileIOProperties.ENDPOINT] = endpoint
+                    }
+                }
+
+        return when (catalogConfig) {
+            is NessieCatalogConfiguration -> {
+                // Nessie relies on the AWS region being set as a system property.
+                System.setProperty("aws.region", config.s3BucketConfiguration.s3BucketRegion.region)
+
+                val nessieProperties =
+                    mutableMapOf(
+                        ICEBERG_CATALOG_TYPE to ICEBERG_CATALOG_TYPE_NESSIE,
+                        URI to catalogConfig.serverUri,
+                        NessieConfigConstants.CONF_NESSIE_REF to
+                            icebergCatalogConfig.mainBranchName,
+                        WAREHOUSE_LOCATION to icebergCatalogConfig.warehouseLocation,
                     )
+
+                // Add optional Nessie auth token if provided.
+                catalogConfig.accessToken?.let { token ->
+                    nessieProperties[NessieConfigConstants.CONF_NESSIE_AUTH_TYPE] = "BEARER"
+                    nessieProperties[NessieConfigConstants.CONF_NESSIE_AUTH_TOKEN] = token
                 }
+
+                nessieProperties + s3CommonProperties
+            }
+            is GlueCatalogConfiguration -> {
+                val clientCredentialsProviderPrefix =
+                    AwsClientProperties.CLIENT_CREDENTIALS_PROVIDER + "."
+
+                val glueProperties =
+                    mapOf(
+                        ICEBERG_CATALOG_TYPE to ICEBERG_CATALOG_TYPE_GLUE,
+                        WAREHOUSE_LOCATION to icebergCatalogConfig.warehouseLocation,
+                        AwsProperties.GLUE_CATALOG_ID to catalogConfig.glueId,
+                        AwsClientProperties.CLIENT_REGION to
+                            config.s3BucketConfiguration.s3BucketRegion.region,
+                        AwsClientProperties.CLIENT_CREDENTIALS_PROVIDER to
+                            GlueCredentialsProvider::class.java.name,
+                        "${clientCredentialsProviderPrefix}access-key-id" to awsAccessKeyId,
+                        "${clientCredentialsProviderPrefix}secret-access-key" to awsSecretAccessKey
```

Comment:
Nit:  we can probably turn `access-key-id` and `secret-access-key` to constants, given that they are also referenced in `GlueCredentialsProvider` also.  Better to have one place to update if/when it changes.